### ISSUE TITLE
Consider splitting operations into Core and Higher level definitions

### ISSUE THREAD

**Issue by tplooker** [decentralized-identity/bbs-signature#131]
As raised by @BasileiosKal previously, to improved readability around the defined operations we could elect to group the operations into core and non-core, for instance `Sign`, `Verify`, `ProofGen` and `ProofVerify` could become `CoreSign`, `CoreVerify`, `CoreProofGen` and `CoreProofVerify`. We could then define more application specific higher-level operations like `SingleSign` and `MultiSign` where the function signatures could be something like the following

```
signature = SingleSign(SK, header)
```

```
signature = MultiSign(SK, header, (msg_1,..., msg_L))
```

**Note** these ^ are just ideas about how we could encapsulate common usage patterns.

**Comment by tplooker** [comment]
Some of this is being addressed by #154

**Comment by tplooker** [comment]
Closing with the merge of #154 which settled the restructure of the sections
